modules/apps/headless/headless-delivery/headless-delivery-test/src/testFunctional/macros/MessageBoardMessageAPI.macro

```diff
@@ -0,0 +1,70 @@
+definition {
+
+	macro _curlMessageBoardMessage {
+		var portalURL = JSONCompany.getPortalURL();
+
+		if (isSet(messageBoardThreadId)) {
+			var api = "message-board-threads/${messageBoardThreadId}/message-board-messages";
+		}
+		else if (isSet(messageBoardMessageId)) {
+			var api = "message-board-messages/${messageBoardMessageId}";
+		}
+		else {
+			var api = "message-board-messages/${parentMessageBoardMessageId}/message-board-messages";
+		}
+
+		var command = '''
+				-u test@liferay.com:test \
+				-H Content-Type: application/json
+		''';
+
+		if (isSet(headline)) {
+			var body = '''
+				-d {
+					"headline": "${headline}"
+				}
+			''';
+
+			var command = StringUtil.add(${command}, " \ ${body}", "");
+		}
+
+		var curl = StringUtil.add("${portalURL}/o/headless-delivery/v1.0/${api}", "\ ${command}", "");
+
+		return ${curl};
+	}
+
+	macro getIdOfCreatedChildMbMessageByMbMessageId {
+		Variables.assertDefined(parameterList = "${headline},${parentMessageBoardMessageId}");
+
+		var curl = MessageBoardMessageAPI._curlMessageBoardMessage(
+			headline = ${headline},
+			parentMessageBoardMessageId = ${parentMessageBoardMessageId});
+
+		var mbMessageId = JSONCurlUtil.post(${curl}, "$.id");
+
+		return ${mbMessageId};
+	}
+
+	macro getIdOfCreatedMbMessageByMbThreadId {
```

Comment:
if we don't have any other methods for getting ID maybe we can shorten the title? Also, the needed parameters are listed in parameterList
this method also does two things, one is to create the messsage board and another one is to get the ID, maybe we need to start dividing them to be the methods more generic

Reply:
Updated.